### ISSUE TITLE
Mesh Structure Computations Fail on Device Cuda:1

### ISSUE THREAD

**Issue by argupta98** [facebookresearch/pytorch3d#157]
Hello,

Thanks for making this awesome library!

 I have been trying to get some code working for computing error between a Mesh Structure and PointCloud using `pytorch3d.loss.point_mesh_face_distance`; however, I was consistently getting `0.0` as the loss when I know that the mesh error should be higher. Upon checking the mesh by computing area, I was also getting `0.0` consistently. The reason is that I was on `torch.device(cuda:1)`. On  `cuda:0` everything works as expected. 

Here is the minimal example:
```python
import torch
import pytorch3d
import pytorch3d.structures
import pytorch3d.loss


if __name__ == "__main__":
    vertices_1 = torch.randn(size=(100, 3), dtype=torch.float32)
    vertices_2 = torch.randn(size=(100, 3), dtype=torch.float32)
    faces_1 = torch.randint(high=99, size=(50, 3)).long()
    faces_2 = torch.randint(high=99, size=(50, 3)).long()
    pointcloud_1 = torch.randn(size=(1000, 3), dtype=torch.float32)
    pointcloud_2 = torch.randn(size=(1000, 3), dtype=torch.float32)
    mesh = pytorch3d.structures.Meshes([vertices_1, vertices_2], [faces_1, faces_2]).to(torch.device("cuda:1"))
    mesh_area = mesh.faces_areas_packed().sum()
    pcd = pytorch3d.structures.Pointclouds([pointcloud_1, pointcloud_2]).to(torch.device("cuda:1"))
    print("mesh loss: {}".format(pytorch3d.loss.point_mesh_face_distance(mesh, pcd)))
    print("mesh area: {}".format(mesh_area)
```
If `cuda:0`, I get:
```
mesh loss: 0.1963171511888504
mesh area: 188.90768432617188
```

If changed to `cuda:1`, I get:

```
mesh loss: 0.0
mesh area: 1.401298464324817e-43
```

I get this consistently, so it is not some fluke from using random.

Could this either be fixed, or at least have some error upon moving to `cuda:1` so that others know that this functionality cannot be supported?

Thanks!

**Comment by gkioxari** [comment]
Can you confirm that you do the following?
```
device = torch.device("cuda:1")
torch.cuda.set_device(device)
```

**Comment by gkioxari** [comment]
I just reproduced your code and with explicitly setting the device it works. We will add support to the our source kernels to launch in the device of the tensors without having to explicitly set the devices in the future.
